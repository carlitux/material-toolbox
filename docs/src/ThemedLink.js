// @flow
import * as React from 'react';

/* eslint-disable */
import Theme from 'react-mcw/theme';
/* eslint-enable */

type Props = { to: string, children: React.Node };

const ThemedLink = ({ to, children, ...rest }: Props) => (
  <Theme textStyle="primary">
    <a {...rest} href={to}>
      {children}
    </a>
  </Theme>
);

export default ThemedLink;
