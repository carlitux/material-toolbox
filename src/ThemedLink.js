// @flow
import * as React from 'react';

import Theme from './material-toolbox/theme';

type Props = {
  to: string,
  children: React.Node,
  component: React.ComponentType<any> | string,
};

const ThemedLink = ({ to, children, component: Component, ...rest }: Props) => {
  const props = {
    href: Component === 'a' ? to : undefined,
    to: Component !== 'a' ? to : undefined,
  };
  return (
    <Theme textStyle="primary">
      <Component {...rest} {...props}>
        {children}
      </Component>
    </Theme>
  );
};

ThemedLink.defaultProps = {
  component: 'a',
};

export default ThemedLink;
