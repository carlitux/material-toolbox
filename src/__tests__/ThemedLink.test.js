import React from 'react';
import { shallow } from 'enzyme';
import ThemedLink from '../ThemedLink';

describe('component::ThemedLink', () => {
  it('Should match snapshoot', () => {
    const component = shallow(<ThemedLink />);
    expect(component).toMatchSnapshot();
  });
});
