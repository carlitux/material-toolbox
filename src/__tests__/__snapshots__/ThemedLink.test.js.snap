// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component::ThemedLink Should match snapshoot 1`] = `
<Theme
  textStyle="primary"
>
  <a />
</Theme>
`;
